tests/BridgeImplementationTest.php

```diff
@@ -1,188 +1,204 @@
 <?php
-
-use PHPUnit\Framework\TestCase;
-use PHPUnit\Framework\TestResult;
-use PHPUnit\Framework\AssertionFailedError;
-
 require_once __DIR__ . '/../lib/rssbridge.php';
 
-/**
- * This class checks bridges for implementation details:
- *
- * - A bridge must not implement public functions other than the ones specified
- *   by the bridge interfaces. Custom functions must be defined in private or
- *   protected scope.
- * - getName() must return a valid string (non-empty)
- * - getURI() must return a valid URI
- * - A bridge must define constants for NAME, URI, DESCRIPTION and MAINTAINER,
- *   CACHE_TIMEOUT and PARAMETERS are optional
- */
-final class BridgeImplementationTest extends TestCase {
-	private function CheckBridgePublicFunctions($bridgeName){
-
-		$parent_methods = array();
-
-		if(in_array('BridgeInterface', class_parents($bridgeName))) {
-			$parent_methods = array_merge($parent_methods, get_class_methods('BridgeInterface'));
-		}
-
-		if(in_array('BridgeAbstract', class_parents($bridgeName))) {
-			$parent_methods = array_merge($parent_methods, get_class_methods('BridgeAbstract'));
-		}
-
-		if(in_array('FeedExpander', class_parents($bridgeName))) {
-			$parent_methods = array_merge($parent_methods, get_class_methods('FeedExpander'));
-		}
-
-		// Receive all non abstract methods
-		$methods = array_diff(get_class_methods($bridgeName), $parent_methods);
-		$method_names = implode(', ', $methods);
+use PHPUnit\Framework\TestCase;
 
-		$errmsg = $bridgeName
-		. ' implements additional public method(s): '
-		. $method_names
-		. '! Custom functions must be defined in private or protected scope!';
+class BridgeImplementationTest extends TestCase {
+	const PATH_BRIDGES = __DIR__ . '/../bridges/';
 
-		$this->assertEmpty($method_names, $errmsg);
+	private $class;
+	private $obj;
 
+	/**
+	 * @dataProvider dataBridgesProvider
+	 */
+	public function testClassName($path) {
+		$this->setBridge($path);
+		$this->assertTrue($this->class === ucfirst($this->class), 'class name must start with uppercase character');
 	}
 
-	private function CheckBridgeGetNameDefaultValue($bridgeName){
-
-		if(in_array('BridgeAbstract', class_parents($bridgeName))) { // Is bridge
-
-			if(!$this->isFunctionMemberOf($bridgeName, 'getName'))
-				return;
-
-			$bridge = new $bridgeName();
-			$abstract = new BridgeAbstractTest();
-
-			$message = $bridgeName . ': \'getName\' must return a valid name!';
-
-			$this->assertNotEmpty(trim($bridge->getName()), $message);
-
-		}
-
+	/**
+	 * @dataProvider dataBridgesProvider
+	 */
+	public function testClassType($path) {
+		$this->setBridge($path);
+		$this->assertInstanceOf('BridgeAbstract', $this->obj);
```

Comment:
You can use a different syntax here to make the intent clear. Also, brides must implement `BridgeInterface` (`BridgeAbstract` and `FeedExpander` are just simpler to use).

```suggestion
		$this->assertInstanceOf(BridgeInterface::class, $this->obj);
```